src/hotspot/share/opto/loopPredicate.cpp

```diff
@@ -249,7 +250,7 @@ void PhaseIdealLoop::clone_concrete_loop_predicates(Deoptimization::DeoptReason
     uncommon_proj = iff->proj_out(1 - predicate->as_Proj()->_con);
     if (uncommon_proj->unique_ctrl_out() != rgn)
       break;
-    if (iff->is_RangeCheck()) {
+    if (iff->in(1)->Opcode() == Op_Opaque4 && skeleton_predicate_has_opaque(iff)) {
       // Only need to clone range check predicates as those can be changed and duplicated by inserting pre/main/post loops
```

Comment:
Is the comment still correct? It is still always a range check predicate?

Reply:
Hi Nils,
Yes, the comment is still correct. Skeleton predicates are always range checks. But in some cases to protect against overflow, predicates perform range checks with long computations in which case they use an IfNode and not a RangeCheckNode.